kcidb/test_oo.py

```diff
@@ -0,0 +1,192 @@
+                        "origin": "kernelci"
+                    },
+                    {
+                        "build_id": "redhat:redhat.org:619c65f970"
+                                    "9de72e90f2efd0",
+                        "id": "redhat:redhat.org:b9d8be63bc2abca63165"
+                              "de5fd74f0f6d2f0b0d1c",
+                        "origin": "redhat",
+                        "waived": True,
+                        "status": "DONE"
+                    },
+                ]
+            }
+        )
+        self.client = kcidb.oo.Client(source)
+
+    def test_client_variable(self):
+        """Check instance for client variable"""
+        self.assertIsInstance(self.client, kcidb.oo.Client)
+
+    def test_returned_instance(self):
+        """Check that the correct instance is returned from the OO Client."""
+
+        self.assertEqual(self.query_str(""), {})
+
+        self.check_type_instance(self.query_str(
+            ">test[kernelci:kernelci.org:619c656de1fb4af479f2efaa]#"),
+            "test", kcidb.oo.Test
+        )
+
+        self.check_type_instance(self.query_str(
+            ">build[kernelci:kernelci.org:619c6a30636e8fe042f2efa0]#"),
+            "build", kcidb.oo.Build
+        )
+
+        self.check_type_instance(self.query_str(
+            ">checkout[_:redhat:5acb9c2a7bc836e9619c65f9709de72e90f2efd0]#"),
+            "checkout", kcidb.oo.Checkout
+        )
+
+        self.check_type_instance(self.query_str(
+            '>revision["5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1", ""]#'),
+            'revision', kcidb.oo.Revision
+        )
+
+        self.check_type_instance(self.query_str(
+            ">test[kernelci:kernelci.org:619c656de1fb4af479f2efac;"
+            "kernelci:kernelci.org:619c656de1fb4af479f2efae]#"),
+            "test", kcidb.oo.Test
+        )
+
+        self.check_type_instance(self.query_str(
+            ">build[kernelci:kernelci.org:619c6c73764403e3d4f2eff7;"
+            "kernelci:kernelci.org:619c6c8d764403e3d4f2efff]#"),
+            "build", kcidb.oo.Build
+        )
+
+        self.check_type_instance(self.query_str(
+            ">checkout[_:kernelci:5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1;"
+            "_:redhat:5acb9c2a7bc836e9619c65f9709de72e90f2efd0]#"),
+            "checkout", kcidb.oo.Checkout
+        )
+
+        self.check_type_instance(self.query_str(
+            '>revision["5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1", "";'
+            '"bd355732283c23a365f7c55206c0385100d1c389", ""]#'),
+            'revision', kcidb.oo.Revision
+        )
+
+        self.check_data(self.query_str(
+            '>revision["bd355732283c23a365f7c55206c0385100d1c389", ""]#'
+            '>checkout#>build#>test#')
+        )
+
+        self.check_data(self.query_str(
+            '>revision["bd355732283c23a365f7c55206c0385100d1c389", "";'
+            '"5acb9c2a7bc836e9619c65f9709de72e90f2efd0", ""]#'
+            '>checkout#>build#>test#')
+        )
```

Comment:
I see that these are trying to check for multiple objects, but they don't actually test how many and which types are returned.